### ISSUE TITLE
Suggestions/enhacements

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sonuarya/csipsimple#1785]
```
First, I'd like say that this is a wonderful SIP app for Android... but of 
course there's always room for improvement.
Here are a couple of suggestions:

1. Using my new Samsung Galazy Blaze (similar to the S2 just smaller), when I 
activate bluetooth there's no audio either way and after disabling bluetooth 
still no audio on the handset or speakerphone/ringer. Also it freezes the app 
and need to reboot.

2. A simple voice mail feature would be great to pick up calls after a certain 
number of rings.

3. CallerID should have 2 selectable methods to be compatible with more sip 
providers:
Your method just sends the callerID number in the display field of the URI like 
this "CallerID" <sip:username@sipprovider.com>
The second method should send the callerID first in the username field upon the 
initial invite like this:

"<blank>" <sip:CallerID@sipprovider.com>
then on the authentication invite should send this:
"<blank>" <sip:username@sipprovider.com>

Basically on the initial invite we send the callerID in the username field, 
then send the username in the username field for authentication to complete the 
call.

I've been using Vitelity.net for years which is great commercial grade provider 
for just pennies per minute and it uses the second callerID method.

Thanks,
Bill

```

Original issue reported on code.google.com by `procom...@gmail.com` on 15 Jun 2012 at 12:29
